backend/internal/pkg/services/parkingspot/parkingspot_test.go

```diff
@@ -0,0 +1,335 @@
+package parkingspot
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/models"
+	"github.com/ParkWithEase/parkeasy/backend/internal/pkg/repositories/parkingspot"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type mockRepo struct {
+	mock.Mock
+}
+
+const (
+	testOwnerID    = 0
+	testNonOwnerID = 1
+)
+
+var publicTestEntry = parkingspot.Entry{
+	ParkingSpot: models.ParkingSpot{
+		Location: sampleLocation,
+	},
+	InternalID: 0,
+	Owner:      testOwnerID,
+	IsPublic:   true,
+}
+
+var privateTestEntry = parkingspot.Entry{
+	ParkingSpot: models.ParkingSpot{
+		Location: sampleLocation,
+	},
+	InternalID: 0,
```

Comment:
Consider changing the InternalID here to not be confused with the above parking spot